src/core.ts

```diff
@@ -255,10 +263,15 @@ function init() {
   startTime = getUnixTimestamp();
   activePlugins = [];
   bindings = {};
-  nextPayload = [];
   droppedPayloads = {};
-  nextPayloadLength = 0;
   sentBytesCount = 0;
+  compressionWorker = createCompressionWorker(envelope(), onWorkerMessage);
+
+  // If critical API is missing, don't activate Clarity
+  if (!checkFeatures()) {
+    teardown();
+    return false;
+  }
```

Comment:
We should probably do this first in the init, no need to create guids/workers just to tear them down right after

Reply:
In case check features fails, we are logging 'missing features' instrumentation and 'teardown' events. We need variables above to do that.

---
In reply to: [136670454](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136670454) [](ancestors = 136670454)